tests/components/shelly/test_config_flow.py

```diff
@@ -116,6 +147,68 @@ async def test_form_errors_test_connection(hass, error):
     assert result2["errors"] == {"base": base_error}
 
 
+async def test_form_already_configured(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    entry = MockConfigEntry(
+        domain="shelly", unique_id="test-mac", data={"host": "0.0.0.0"}
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "aioshelly.get_info",
+        return_value={"mac": "test-mac", "type": "SHSW-1", "auth": False},
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"host": "1.1.1.1"},
+        )
+
+        assert result2["type"] == "abort"
+        assert result2["reason"] == "already_configured"
+
+    # Test config entry got updated with latest IP
+    assert entry.data["host"] == "1.1.1.1"
+
+
+@pytest.mark.parametrize(
+    "error",
+    [
+        (aiohttp.ClientResponseError(Mock(), (), status=400), "cannot_connect"),
+        (aiohttp.ClientResponseError(Mock(), (), status=401), "invalid_auth"),
+        (asyncio.TimeoutError, "cannot_connect"),
+        (ValueError, "unknown"),
+    ],
+)
+async def test_form_auth_errors_test_connection(hass, error):
+    """Test we handle errors in authenticated devices."""
+    exc, base_error = error
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("aioshelly.get_info", return_value={"mac": "test-mac", "auth": True}):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"host": "1.1.1.1"},
+        )
+
+    with patch(
+        "aioshelly.Device.create",
+        side_effect=exc,
+    ):
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"],
+            {"username": "test username", "password": "test password"},
+        )
+    assert result3["type"] == "form"
+    assert result3["errors"] == {"base": base_error}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)